src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -10,64 +10,166 @@
 import java.nio.ByteBuffer;
 import java.time.Duration;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
 import org.eclipse.jetty.http.HttpStatus;
-import org.junit.AfterClass;
+import org.eclipse.jetty.websocket.api.UpgradeException;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
 
+import com.microsoft.azure.relay.HybridConnectionListener.ControlConnection;
+
 public class HybridConnectionListenerTest {
-	// Max # simultaneous client connections = # of cores - 1
-	// Because one thread need to be reserved for listener
 	private static final int MAX_CONNECTIONS_COUNT = Math.max(1, Runtime.getRuntime().availableProcessors() * 2);
-	private static HybridConnectionListener listener;
+	private static final AutoShutdownScheduledExecutor EXECUTOR = AutoShutdownScheduledExecutor.Create();
 	private static TokenProvider tokenProvider;
 	private static HybridConnectionClient client;
 	private static URI CONNECTION_URI;
+	private static HybridConnectionListener listener;
```

Comment:
>static [](start = 9, length = 6)

does this need to be static?

Reply:
it doesn't have to be, but why not?

---
In reply to: [262215879](https://github.com/Azure/azure-relay-java/pull/28#discussion_r262215879) [](ancestors = 262215879)

Reply:
Because if you instantiate two instances of this class and call beforeEachTest on both of them they should both use the same (last) listener.

---
In reply to: [262227695](https://github.com/Azure/azure-relay-java/pull/28#discussion_r262227695) [](ancestors = 262227695,262215879)